x-pack/plugins/observability/server/routes/rules.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import * as t from 'io-ts';
+import { isoToEpochRt, toNumberRt } from '@kbn/io-ts-utils';
+import Boom from '@hapi/boom';
+import { createObservabilityServerRoute } from './create_observability_server_route';
+import { createObservabilityServerRouteRepository } from './create_observability_server_route_repository';
+import { getTopAlerts } from '../lib/rules/get_top_alerts';
+
+const alertsListRoute = createObservabilityServerRoute({
+  endpoint: 'GET /api/observability/rules/alerts/top',
+  options: {
+    tags: [],
+  },
+  params: t.type({
+    query: t.intersection([
+      t.type({
+        start: isoToEpochRt,
+        end: isoToEpochRt,
+      }),
+      t.partial({
+        kuery: t.string,
+        size: toNumberRt,
+      }),
+    ]),
+  }),
+  handler: async ({ ruleRegistry, context, params }) => {
+    const ruleRegistryClient = ruleRegistry.createScopedRuleRegistryClient({
+      context,
+    });
+
+    if (!ruleRegistryClient) {
+      throw Boom.failedDependency();
+    }
+
+    const {
+      query: { start, end, kuery, size = 100 },
+    } = params;
+
+    return getTopAlerts({
+      ruleRegistryClient,
+      start,
+      end,
+      kuery,
+      size,
+    });
+  },
+});
+
+const alertsDynamicIndexPatternRoute = createObservabilityServerRoute({
+  endpoint: 'GET /api/observability/rules/alerts/dynamic_index_pattern',
+  options: {
+    tags: [],
+  },
+  handler: async ({ ruleRegistry, context }) => {
+    const ruleRegistryClient = ruleRegistry.createScopedRuleRegistryClient({
+      context,
+    });
+
+    if (!ruleRegistryClient) {
+      throw Boom.failedDependency();
+    }
+
+    return ruleRegistryClient.getDynamicIndexPattern();
+  },
+});
```

Comment:
you can use `data.indexPatterns.getFieldsForWildcard` on client, i don't it make sense to create a separate route for it.

Reply:
This call is done with the internal user. I assume that's not possible in the browser?

Reply:
you are right, but will it make a different in this case? AFAIK, you can't assign individual permissions to index patterns. Kibana permissions are weird as far as data goes, they are useful for limiting UI features though.

Reply:
I'm not sure what you mean with "assign individual permissions to index patterns"?